### PR TITLE
refactor(api): user token db query

### PR DESCRIPTION
**Pull request by moT01**
This is a change we discussed in the meeting today - will hopefully cut down on those sentry errors?

Checklist:

- [x] I have read and followed the [contribution guidelines](https://contribute.freecodecamp.org).
- [x] I have read and followed the [how to open a pull request guide](https://contribute.freecodecamp.org/#/how-to-open-a-pull-request).
- [x] My pull request targets the `main` branch of freeCodeCamp.
- [x] I have tested these changes either locally on my machine, or GitPod.

Closes

### CODE REVIEW

File: api-server/src/server/boot/user.js [hunk 1]
```diff
@@ -93,21 +93,23 @@ function deleteUserTokenResponse(req, res) {
...
-    });
-
-    const userToken = userTokenArr[0]?.id;
     let encodedUserToken;
 
-    // only encode if a userToken was found
-    if (userToken) {
-      encodedUserToken = encodeUserToken(userToken);
+    if (queryUser && queryUser.id) {
+      // if queryUser is undefined, this returns the first document
```
Comment (naomi-lgbt) [inline api-server/src/server/boot/user.js:104]: I'm confused by this comment - if `queryUser` is undefined, this query wouldn't execute at all?

Comment (moT01) [reply api-server/src/server/boot/user.js:104]: I added the `if (queryUser...)` cause the db query returns the first doc if you try to use undefined as the `userId`: `where: { userId: queryUser.id }`

Comment (ojeytonwilliams) [reply api-server/src/server/boot/user.js:104]: How about

```js
    const userId = queryUser?.id;
    const userToken = userId
      ? await UserToken.findOne({
          where: { userId }
        })
      : null;

    const encodedUserToken = userToken
      ? encodeUserToken(userToken.id)
      : undefined;
```

I think this code's fairly self-explanatory.

Comment (moT01) [reply api-server/src/server/boot/user.js]: ✅

Comment (ShaunSHamilton) [reply api-server/src/server/boot/user.js]:
```javascript
const encodedUserToken =  queryUser?.id ? await UserToken.findOne({where: {userId: queryUser.id}}) ? encodeUserToken(queryUser?.id ? await UserToken.findOne({where: {userId: queryUser.id}}).id : null) : undefined
```

Comment (naomi-lgbt) [reply api-server/src/server/boot/user.js]: Don't make me remove your maintainer access Shaun

**Review by naomi-lgbt — APPROVED** [review]

**Review by ojeytonwilliams — APPROVED** [review]
LGTM :+1: 

@raisedadead do you want to give this a quick before we deploy?

**Review by ShaunSHamilton — APPROVED** [review]
Fine, I will approve this 👍

**Review by raisedadead — APPROVED** [review]
I am going ahead and deploying this on staging. I would like for all of us to go ahead and test it thru and thru.

cc @freeCodeCamp/dev-team

**Pull request merged**